[zuohui48] ip

zuohui48 opened the pull request:
# MrDuke

> "All models are wrong, but some are useful." -George Box

MrDuke helps you to micromanage all your tasks. It's
- easy to use 
- text-based
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

You can: 
- [x] Manage tasks
- [x] Manage deadlines 
- [x] Set reminders


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

zuohui48 committed:
Level 1. Greet, Echo, Exit

zuohui48 committed:
Level 2. Add, List

zuohui48 committed:
Level 3. Mark as Done

zuohui48 committed:
Level 4. ToDos, Events, Deadlines

zuohui48 committed:
Automated Text UI Testing

zuohui48 committed:
Handle errors

zuohui48 committed:
enable deletion of tasks

zuohui48 committed:
enable deletion of tasks

zuohui48 committed:
Save and load data from Duke.txt

zuohui48 committed:
Add date and time

zuohui48 committed:
Merge branch 'branch-Level-8'

zuohui48 committed:
Fix loading data

zuohui48 committed:
Implement more OOP

shaowi commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,41 @@
+    }
+    public boolean checkDelete(String input) {
+        return input.split(" ")[0].equals("delete");
+    }
```
Boolean variables/methods should be named to sound like booleans.

shaowi commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,105 @@
+import java.io.File;
+import java.io.IOException;
+import java.io.FileWriter;
+import java.io.*;
```
Imported classes should always be listed explicitly.

shaowi commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,41 @@
+import java.util.Arrays;
+import java.util.stream.Stream;
+public class Parser {
+    private static String[] VALID_COMMANDS = {"mark", "unmark", "list", "delete", "bye", "todo", "event", "deadline"};
```
final keyword can be specified for constants

shaowi commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,136 @@
+import java.util.ArrayList;
+public class TaskList {
+    private ArrayList<Task> list;
```
Possible to be more clear of what list is this ? :)

zuohui48 committed:
add packages

zuohui48 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

zuohui48 committed:
Use Gradle

zuohui48 committed:
Add folders

zuohui48 committed:
add JUnit tests for Task and Deadline

zuohui48 committed:
Add JUnit tests for Task and Deadline

zuohui48 committed:
update user input logic

zuohui48 committed:
Add JavaDoc comments.

zuohui48 committed:
Follow the coding standard

zuohui48 committed:
Add find keyword funtionality

zuohui48 committed:
Merge branch 'branch-A-JavaDoc'

zuohui48 committed:
Merge branch 'branch-Level-9'

zuohui48 committed:
Use CheckStyle

zuohui48 committed:
Use CheckStyle

shaowi reviewed (commented):
Looks good to merge!

zuohui48 committed:
Implement GUI

zuohui48 committed:
Implement GUI

zuohui48 committed:
Fix task tests

zuohui48 committed:
Update error messages when loading data

zuohui48 committed:
Use assert feature

There are important assumptions that should hold at various points
in the code.

The assert feature is use to ensure that none of these assumptions
are being violeted.

zuohui48 committed:
Handle user input with Parser

zuohui48 committed:
Merge pull request #1 from zuohui48/branch-A-CodeQuality

Handle user input with Parser

zuohui48 committed:
Merge branch 'master' into branch-A-Assertions

zuohui48 committed:
Merge pull request #2 from zuohui48/branch-A-Assertions

Use assert feature

zuohui48 committed:
Handle adding of duplicate tasks

zuohui48 committed:
Add checks for structure of inputs

zuohui48 committed:
Add Ui.png

zuohui48 committed:
Clean up duke.txt

zuohui48 committed:
Add a user guide

zuohui48 committed:
Add user guide

zuohui48 committed:
Update Duke replies for invalid commands

zuohui48 committed:
Update .jar file name

zuohui48 committed:
Update .jar file name

zuohui48 committed:
Update duke.txt path

zuohui48 committed:
Add SLAP to long methods

zuohui48 committed:
Remove dead code

zuohui48 committed:
Improve find command

zuohui48 committed:
Remove dead code

zuohui48 committed:
Fix long response messages getting cut off
